app/routes.py

```diff
@@ -1,97 +1,232 @@
 def get_all_harvest_sources():
-    result = db.get_all_harvest_sources()
+    source = db.get_all_harvest_sources()
+    org = db.get_all_organizations()
+    return render_template("harvest_source.html", sources=source, organizations=org)
+
+@mod.route("/harvest_source/", methods=["GET"])    
+@mod.route("/harvest_source/<source_id>", methods=["GET"])
+def get_harvest_source(source_id=None):
+    if source_id:
+        source = db.get_harvest_source(source_id)
+        return jsonify(source) if source  else ("Not Found", 404)
+
+    organization_id = request.args.get("organization_id")
+    if organization_id:
+        source = db.get_harvest_source_by_org(organization_id)
+        if not source:
+            return "No harvest sources found for this organization", 404
+    else:
+        source = db.get_all_harvest_sources()
+    return jsonify(source)
+        
+@mod.route("/harvest_source/<source_id>", methods=["PUT"])
+def update_harvest_source(source_id):
+    result = db.update_harvest_source(source_id, request.json)
     return result
 
-@mod.route('/harvest_jobs', methods=['GET'])
-def get_all_harvest_jobs():
-    result = db.get_all_harvest_jobs()
+@mod.route("/harvest_source/<source_id>", methods=["DELETE"])
+def delete_harvest_source(source_id):
+    result = db.delete_harvest_source(source_id)
     return result
 
-@mod.route('/harvest_errors_by_job/<job_id>', methods=['GET'])
-def get_all_harvest_errors_by_job(job_id):
-    try:
-        result = db.get_all_harvest_errors_by_job(job_id)
-        return result
-    except Exception:
-        return " provide job_id"
+@mod.route("/harvest_job/add", methods=["POST"])
+def add_harvest_job():
+    if request.is_json:
+        job = db.add_harvest_job(request.json)
+        if job:
+            return jsonify({"message": f"Added new harvest job with ID: {job.id}"})
+        else:
+            return jsonify({"error": "Failed to add harvest job."}), 400
+    else:
+        return jsonify({"Please provide harvest job with json format."})
 
-@mod.route('/harvest_source/<source_id>', methods=['GET'])
-def get_harvest_source(source_id):
+@mod.route("/harvest_job/", methods=["GET"])    
+@mod.route("/harvest_job/<job_id>", methods=["GET"])
+def get_harvest_job(job_id=None):
     try:
-        result = db.get_harvest_source(source_id)
-        return result
+        if job_id:
+            job = db.get_harvest_job(job_id)
+            return jsonify(job) if job else ("Not Found", 404)
+
+        source_id = request.args.get("harvest_source_id")
+        if source_id:
+            job = db.get_harvest_job_by_source(source_id)
+            if not job:
+                return "No harvest jobs found for this harvest source", 404
+        else:
+            job = db.get_all_harvest_jobs()
+    
+        return jsonify(job)
     except Exception:
-        return " provide source_id"
+        return "Please provide correct job_id or harvest_source_id"
```

Comment:
remove `harvest_source_id`

Reply:
since we may get job either with job_id or source_id, the message is for both possible situations.